daml-assistant/exe/DAML/Assistant.hs

```diff
@@ -69,15 +71,56 @@ main = displayErrors $ do
     case userCommand of
 
         Builtin Version -> do
-            putStr . unlines $
-              [ "SDK version: "
-              <> maybe "unknown" versionToString envSdkVersion
-              , "Latest stable release: "
-              <> maybe "unknown" versionToString envLatestStableSdkVersion
-              , "Assistant version: "
-              <> maybe "unknown" (versionToString . unwrapDamlAssistantSdkVersion)
-                 envDamlAssistantSdkVersion
-              ]
+            installedVersionsE <- tryAssistant $ getInstalledSdkVersions envDamlPath
+            defaultVersionM <- tryAssistantM $ getDefaultSdkVersion envDamlPath
+
+            let asstVersion = unwrapDamlAssistantSdkVersion <$> envDamlAssistantSdkVersion
+                envVersions = catMaybes
+                    [ envSdkVersion
+                    , envLatestStableSdkVersion
+                    , asstVersion
+                    ]
+
+                isInstalled =
+                    case installedVersionsE of
+                        Left _ -> const True
+                        Right vs -> (`elem` vs)
+
+                versionAttrs v = catMaybes
+                    [ "active"
+                        <$ guard (Just v == envSdkVersion)
+                    , "default"
+                        <$ guard (Just v == defaultVersionM)
+                    , "assistant"
+                        <$ guard (Just v == asstVersion)
+                    , "latest release"
+                        <$ guard (Just v == envLatestStableSdkVersion)
+                    , "not installed"
+                        <$ guard (not (isInstalled v))
+                    ]
+
+                -- | Workaround for Data.SemVer old unfixed bug (see https://github.com/brendanhay/semver/pull/6)
+                -- TODO: move away from Data.SemVer...
+                versionCompare v1 v2 =
+                    if v1 == v2
+                        then EQ
+                        else compare v1 v2
+
+                versions = nubSortBy versionCompare (envVersions ++ fromRight [] installedVersionsE)
+                versionTable = [ (versionToText v, versionAttrs v) | v <- versions ]
+                versionWidth = maximum (1 : map (T.length . fst) versionTable)
```

Comment:
Should this padding code exist somewhere else as a utility?

Reply:
Probably! I'll extract it out if/when it becomes useful elsewhere.